### PR TITLE
don't import our crypto shim if crypto is supported natively

### PR DESCRIPTION
**Pull request by alan-codaio**
For environments using Node version >= 19.0.0, running any Coda CLI command would result in the following error:
```
TypeError: Cannot set property crypto of #<Object> which has only a getter
    at ../../../packs/node_modules/@codahq/packs-sdk/dist/testing/injections/crypto_shim.js (/private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/bundle.js:148:21)
    at __init (/private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/bundle.js:15:58)
    at ../../../../../../private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/browserify-bundle.js (/private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/bundle.js:155:7)
    at __require2 (/private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/bundle.js:18:52)
    at /private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/bundle.js:8420:10
    at Object.<anonymous> (/private/var/folders/50/_v53nyjd0vb2r94bzn5nbcqm0000gp/T/coda-packs-4ea73dfd-3391-4b5e-82f7-c91f11182216nGo4p5/bundle.js:8421:3)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
```

This is because, as @ekoleda-codaio pointed out, Node 19 now supports global.crypto natively, and trying to import our shim throws the error. So, for environments with version >= 19.0.0, we skip attempting to import our crypto shim.

Although, we may need to test to make sure our shim and the Node crypto module are compatible before merging/releasing this change.

### CODE REVIEW

**Review by huayang-codaio — COMMENTED** [review]
I thought about this again and got a bit confused when this runs in isolated vm. 

Even if the bundle is built with node 19 (which comes with a native crypto), it may not necessarily mean that the bundle would have crypto in the isolated vm execution context.

If you have node 19 installed locally, can you try run the fake pack and add a formula that uses `crypto.getRandomValues`? I wonder if it works in isolated-vm. 

Otherwise we may need to enforce a node version < 19.

File: testing/injections/crypto_shim.js [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
 //
 // Lastly, we can move this shim to thunk bundle and register it into global from the thunk bundle.
-// It has the same side effect of the shim though. 
+// It has the same side effect of the shim though.
 //
 // please note that this causes a global leak and needs be ignored in some configs.
-global.crypto = crypto;
+// Node 19 has native support for the crypto module.
+const version = typeof process !== 'undefined' ? process.version.substring(1) : ''; // strip out first char from v19.x.x
+if (!semver.valid(version) || semver.compare(version, '19.0.0') < 0) {
```
Comment (ekoleda-codaio) [inline testing/injections/crypto_shim.js:40]: Would it be simpler to just test for the presence of `global.crypto`?

```ts
if (global.crypto === undefined) {
  global.crypto = crypto;
}
```

**Comment by huayang-codaio** [comment]
cool. let's do the optional shim approach then. @alan-codaio 

```
if (!global.crypto?.getRandomValues) {
  global.crypto = crypto;
}
```

**Comment by ekoleda-codaio** [comment]
Is this PR just waiting on a review?

**Review by huayang-codaio — APPROVED** [review]

**Comment by alan-codaio** [comment]
Let me test this again before I merge since it's been a while.

**Comment by alan-codaio** [comment]
ok - it's working: 
```
alanfang:~/code/packs-sdk [af-crypto-nodejs19]$ ts-node cli/coda.ts execute test/packs/fake_v2.ts Random
(node:5270) [DEP0128] DeprecationWarning: Invalid 'main' field in '/Users/alanfang/code/packs-sdk/node_modules/mold-source-map/package.json' of 'mold-source-map.js'. Please either fix that or report it to the module author
(Use `node --trace-deprecation ...` to show where the warning was created)
150
alanfang:~/code/packs-sdk [af-crypto-nodejs19]$ node -v
v19.0.0
```

File: testing/injections/crypto_shim.js [hunk 2]
```diff
@@ -1,37 +1,41 @@
 // ivm doens't have a crypto implementation. since we browserify modules already, this shim implements the browser crypto interface.
 // inspired by https://github.com/kumavis/polyfill-crypto.getrandomvalues/blob/master/index.js
 
-var MersenneTwister = require('mersenne-twister')
+var MersenneTwister = require('mersenne-twister');
+var semver = require('semver');
```
Comment (dweitzman-codaio) [inline testing/injections/crypto_shim.js:5]: Looks like the `semver` import here may be unused?

Comment (jonathan-codaio) [reply testing/injections/crypto_shim.js:5]: Oh, yeah @alan-codaio can this be removed? It's bloating back bundles.

Comment (alan-codaio) [reply testing/injections/crypto_shim.js:5]: yeah — I'll take it out

Comment (alan-codaio) [reply testing/injections/crypto_shim.js:5]: https://github.com/coda/packs-sdk/pull/2469

**Pull request merged**